examples/plot_heterosc_estimation_modelling.py

```diff
@@ -182,22 +182,35 @@
 # approximation i.e. a piecewise linear interpolation/extrapolation based on the binning results.
 # To ensure that only robust statistic values are used in the interpolation, we set a ``min_count`` value at 30 samples.
 
-slope_curv_to_dh_err = xdem.spatialstats.interp_nd_binning(df, list_var_names=['slope', 'maxc'], statistic='nmad', min_count=30)
+unscaled_dh_err_fun = xdem.spatialstats.interp_nd_binning(df, list_var_names=['slope', 'maxc'],
+                                                           statistic='nmad', min_count=30)
 
 # %%
-# The output is an interpolant function of slope and curvature that we can use to estimate the elevation measurement
-# error at any point.
+# The output is an interpolant function of slope and curvature that predicts the elevation error at any point. However,
+# this predicted error might have a spread slightly off from that of the data:
 #
-# For instance:
+# We compare the spread of the elevation difference on stable terrain and the average predicted error:
+dh_err_stable = unscaled_dh_err_fun((slope_arr, maxc_arr))
+
+print('The spread of elevation difference is {:.2f} '
+      'compared to a mean predicted elevation error of {:.2f}.'.format(xdem.spatialstats.nmad(dh_arr),
+                                                                       np.nanmean(dh_err_stable)))
 
-for s, c in [(0.,0.1), (50.,0.1), (0.,20.), (50.,20.)]:
+# %%
+# Thus, we rescale the function to exactly match the spread on stable terrain using the
+# ``xdem.spatialstats.two_step_standardization`` function, and get our final error function.
+
+zscores, dh_err_fun = xdem.spatialstats.two_step_standardization(dh_arr, list_var=[slope_arr, maxc_arr],
+                                                           unscaled_error_fun=unscaled_dh_err_fun)
+
+for s, c in [(0., 0.1), (50., 0.1), (0., 20.), (50., 20.)]:
     print('Elevation measurement error for slope of {0:.0f} degrees, '
-          'curvature of {1:.2f} m-1: {2:.1f}'.format(s, c/100, slope_curv_to_dh_err((s,c)))+ ' meters.')
+          'curvature of {1:.2f} m-1: {2:.1f}'.format(s, c/100, dh_err_fun((s, c)))+ ' meters.')
 
 # %%
-# The same function can be used to estimate the spatial distribution of the elevation measurement error over the area:
+# This function can be used to estimate the spatial distribution of the elevation error on the extent of our DEMs:
 maxc = np.maximum(np.abs(profc), np.abs(planc))
-dh_err = slope_curv_to_dh_err((slope, maxc))
+errors = dh_err_fun((slope, maxc))
 
 plt.figure(figsize=(8, 5))
 plt_extent = [
```

Comment:
Instead of having several lines to get the extent and colorbar etc, you could convert errors to a Raster object and use the function `Raster.show`.

Reply:
This is like this because I copied how Erik plotted the Figures in his first examples. I open an issue so that we correct this everywhere.

Reply:
I think I corrected it in many examples in a previous PR, but maybe not all. I believe there was at least one situation where it was not possible, but I can't remember why.